cli/monitor.py

```diff
@@ -136,20 +129,18 @@ def __init__(self, name, steps, enable_dynamic):
         self._dynamic_steps = {}
 
         self._steps = []
-        _curr_state = None
         for step in self._parsed_steps:
             wrapper = None
             if isinstance(step, SaltRunner):
-                wrapper = Stage.Step(step, step.fun, len(self._steps)+1)
-                _curr_state = None
+                wrapper = Stage.Step(step, step.function, len(self._steps)+1)
             elif isinstance(step, SaltState):
-                wrapper = Stage.TargetedStep(step, step.state, len(self._steps)+1)
-                _curr_state = wrapper
-            elif isinstance(step, SaltModule) or isinstance(step, SaltBuiltIn):
-                assert _curr_state
-                _curr_state.sub_steps.append(Stage.Step(step, step.fun,
-                                                        len(_curr_state.sub_steps)+1))
-                continue
+                wrapper = Stage.TargetedStep(step, step.sls, len(self._steps)+1)
+                for _, s_steps in step.steps.items():
```

Comment:
I could not figure out where `step.steps` comes from, is it a dict? If so, I think it is cleaner to just iterate over its values with `for s_steps in step.steps.values()`

Reply:
`step.steps` comes from https://github.com/SUSE/DeepSea/pull/1149/files#diff-6c2e6839d23a2408ea246962365247edR437 and yes it's a dict